### ISSUE TITLE
isWatching task as preLaunchTask in launch.json

### ISSUE THREAD

**Issue by skolmer** [Microsoft/vscode#6209]
It would be nice to have a watcher task as preLaunchTask in launch.json. At the moment this is not possible because the launch task will not start before the preLaunchTask ends.
A long running preLaunchTask should have a regex similar to problemWatcher to wait for a specific text in the output window before executing the launch task.
In my scenario I'm running webpack dev server as a watcher task. I want this task to run if <kbd>F5</kbd> is hit and the preLaunchTask is not running. To delay the debugger launch it would be useful to test for a specific string in the output window `webpack: bundle is now VALID.`

**Comment by isidorn** [comment]
@skolmer it is possible to have a watcher task as a `preLaunchTask` in `launch.json`. Just checkout the VScode typescript extension template - it has a watching preLaunchTask.
What is your watch task, can you create a small reproducable sample for which we can try to reproduce this.

@dbaeumer I guess it might happen that I am not getting an 'inactive' event from the tasks framework for @skolmer use case

**Comment by skolmer** [comment]
@isidorn [This](https://github.com/skolmer/react-hot-boilerplate-vscode) is the project boilerplate I'm working on. It has a watching task that runs webpack-dev-server. If you set `development` task as a prelauchTask in launch.json and hit <kbd>F5</kbd> you can reproduce the described behavior.

**Comment by isidorn** [comment]
@dbaeumer I double checked and I am not getting the Inactive event from the tasks framework in this case. Forwarding to you

**Comment by dbaeumer** [comment]
@skolmer I see your point. AS a workaround couldn't you define a problemMatcher with a regexp not matching anything. In the problem matcher you then set the watching property to define regular expression that signals when the watching tasks begins and when it ends. An example of such a property for the tsc watching compilation mode is:
```json
        "watching": {
            "activeOnStart": true,
            "beginsPattern": "^\\s*(?:message TS6032:|\\d{1,2}:\\d{1,2}:\\d{1,2} (?:AM|PM) -) File change detected\\. Starting incremental compilation\\.\\.\\.",
            "endsPattern": "^\\s*(?:message TS6042:|\\d{1,2}:\\d{1,2}:\\d{1,2} (?:AM|PM) -) Compilation complete\\. Watching for file changes\\."
        }
```

**Comment by dbaeumer** [comment]
We should support defining watch begin / end pattern without a problem pattern as well.

**Comment by skolmer** [comment]
@dbaeumer Thanks! I wasn't aware of the watching config. [This](https://github.com/skolmer/react-hot-boilerplate-vscode/commit/a066924233cc16e844db0feb25482313605adbad) did exactly what I wanted to achieve. 
```
{
	"owner": "custom",
	"pattern": [],
	"watching": {
		"activeOnStart": true,
		"beginsPattern": "webpack: bundle is now INVALID",
		"endsPattern": "webpack: bundle is now VALID"
	}
}
```
It is showing `The problem pattern is invalid. It must have at least a file, message and line or location match group.` in the output but is working as expected.

**Comment by dbaeumer** [comment]
@skolmer Great to see you got it working. I will keep this issue open to address the comment https://github.com/Microsoft/vscode/issues/6209#issuecomment-218154446

**Comment by skolmer** [comment]
@dbaeumer I couldn't find any documentation about the watching config. Would be nice to have this feature in the docs https://code.visualstudio.com/Docs/editor/tasks#_defining-a-problem-matcher

**Comment by skolmer** [comment]
@dbaeumer My workaround (https://github.com/Microsoft/vscode/issues/6209#issuecomment-218175415) stopped working in one of the latest insider builds. The prelaunch task is now failing. Do you have a suggestion how to fix this?

**Comment by dbaeumer** [comment]
@skolmer how does the task fail?

**Comment by skolmer** [comment]
@dbaeumer This is the error message:
```
Error: Invalid problemMatcher description. A matcher must at least define a pattern, owner and a file location. The problematic matcher is:
{
    "owner": "custom",
    "watching": {
        "activeOnStart": true,
        "beginsPattern": "^webpack: Compiling\\.\\.\\.$",
        "endsPattern": "^webpack: Compiled successfully\\.$"
    }
}
```

**Comment by dbaeumer** [comment]
@skolmer Thanks.

**Comment by dbaeumer** [comment]
@skolmer I looked into the real fix for this and it basically requires some major restructuring I only want to do for the new terminal runner. Basically we need to allow to defined watching patterns on the task level as well.

I found a way how you can make this work right now with a different work around. Define the problem matcher as follows:

```json
			"problemMatcher": {
				"owner": "custom",
				"pattern": { "regexp": "__________" },
				"watching": {
					"activeOnStart": true,
					"beginsPattern": "^webpack: Compiling\\.\\.\\.$",
					"endsPattern": "^webpack: Compiled successfully\\.$"
				}
			}

```

As the regexp use a regular expression that doesn't match anything in the output.

**Comment by dbaeumer** [comment]
As mention the real fix is to allow something like
```json
{
    "taskName": "abc",
    "background": {
         "activeOnStart": true,
         "beginsPattern": "^webpack: Compiling\\.\\.\\.$",
         "endsPattern": "^webpack: Compiled successfully\\.$"
    }
}
```

**Comment by skolmer** [comment]
Thanks for the workaround!

**Comment by dbaeumer** [comment]
Moving to Mai due to shifted priorities.

**Comment by AshleyGrant** [comment]
Another important part of this enhancement is to terminate the preLaunch watch task when the debugging session is ended.

**Comment by dbaeumer** [comment]
@AshleyGrant nice idea. Can you please open a separate issue since this should be done via the debugger. @isidorn FYI.

**Comment by AshleyGrant** [comment]
@dbaeumer done. Another thought I've had regarding this issue is that a check should be done to see if the watching task is already running. In that situation, a duplicate task probably should not be launched.

**Comment by jpap** [comment]
Any recent updates on this? I'm also hitting this using a webpack hot-reload server to debug an electron app. :D

**Comment by niemyjski** [comment]
Any updates on this?

**Comment by jasonnutter** [comment]
You can now run tasks as background tasks: https://code.visualstudio.com/docs/editor/tasks#_background-watching-tasks

**Comment by AshleyGrant** [comment]
This still doesn't really fully cover what the feature request is. Currently, I have to launch the background task and *then* launch a debugging session.

**Comment by jasonnutter** [comment]
I was able to configure VS Code to launch both with the following config:

`tasks.json`
```js
{
    "version": "2.0.0",
    "tasks": [
        {
            "type": "gulp",
            "label": "build",
            "task": "webpack-watch-dev", // gulp webpack-watch-dev is the gulp task I want to run in the background
            "isBackground": true,
            "group": {
                "kind": "build",
                "isDefault": true
            }
        }
    ]
}
```

launch.json
```js
{
    // Use IntelliSense to learn about possible Node.js debug attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "type": "node",
            "request": "launch",
            "name": "Launch Program",
            "program": "${workspaceRoot}/server/index.js",
            "env": {
                "NODE_ENV": "development"
            },
            "preLaunchTask": "build", 
            "console": "integratedTerminal"
        }
     ]
}
````

**Comment by AshleyGrant** [comment]
Well, I stand corrected.. Maybe someone from the VS Code team can look at removing that prompt, or giving a configuration to make it go away.

**Comment by jasonnutter** [comment]
Yeah, I've been meaning to confirm it's an actual bug (and not just something caused by code, as my build task does take a bit to fully start), and file an issue if so.

**Comment by synthecypher** [comment]
@dbaeumer can you explain why your comment works?

More specifically what these parts do and what they are working around?

```json
// ...
  "owner": "custom",
  "pattern": { "regexp": "__________" },
// ...
```

**Comment by dbaeumer** [comment]
Currently you can't define a problem matcher without a pattern. So the pattern I picked is likely not matching anything. Otherwise it would report errors although there aren't any.

**Comment by taylorstutts** [comment]
I cannot get this to work. Am I misunderstanding the intent of the current design?

I am trying to run debugging with chrome from launch.json, running webpack-dev-server, with chrome only launching after the server is running.

Instead, the launch task (Chrome) always executes immediately (before the server is initialized) if the task has isBackground set to true.

I thought I should be able to accomplish this by adding the dummy regex to the problem matcher on webpack-dev-server, and adding the begin/end regex to the background property on the problem matcher. However, the regexes in the problem matcher do not appear to affect anything (aside from that the task cannot be terminated if the main regex is omitted).

Chrome is run from a compound configuration in launch.json:
```
{
   "version": "0.2.0",
   "configurations": [
        {
            // ...
        },
        {
            "name": "Launch_Chrome",
            "type": "chrome",
            "request": "launch",            
            "url": "http://localhost:8080",
            "preLaunchTask": "webpack_dev_server_task",
        }
    ],
    "compounds": [
        {
            "name": "Compound",
            "configurations": [
                // ...
                "Launch_Chrome"
            ]
        }
    ]
}
```
and webpack-dev-server is run from tasks.json:
```
{
    "version": "2.0.0",
    "tasks": [
        {
            "label": "webpack_dev_server_task",
            "command": "webpack-dev-server",
            "dependsOn": [
                "build_webpack_task"
            ],
            "type": "shell",
            "isBackground": true,
            "problemMatcher": {
                "owner": "custom",
                "pattern": {
                    "regexp": "_____"
                },
                "background": {
                    "activeOnStart": false,
                    "beginsPattern": "^.*Project is running at.*$",
                    "endsPattern": "^.*webpack: Compiled successfully..*$"
                },
            }
        }
    ]
}
```

**Comment by qiulang** [comment]
@dbaeumer none of solutions mentioned above work for me (I am using 1.23.1). I have tried 
`"isBackground": true` , `"problemMatcher" regexp  ...`

Still, I can see after my preLaunchTask (to start the webpack server) finished, launch.json did not launch chrome. Only after I killed preLaunchTask did it launch chrome.

**Comment by dbaeumer** [comment]
Does the webpack output contain the lines `Project is running at` and `webpack: Compiled successfully`

**Comment by JohnPascololo** [comment]
@dbaeumer Hello Dirk, has there been any progress on this matter? I'm having a hard time setting up a rather basic configuration with a launch configuration which runs and debugs a .netcore app, as prelaunch I need to call two tasks: one npm task to do build stuff and a second watching task for webpack devserver.

For the watching task I want to use the tsc-watch problem solver to get hints to my code when it does not compile. Is this supposed to get easier at some point in the future?

**Comment by dbaeumer** [comment]
No progress yet. For the watching task: does it only compile or do you want to wait until the wepkack server is running?

**Comment by BertoMejia** [comment]
Did something change in the latest update? In a previous project, I was able to get my angular front end task to run and then debug the back end using the following tasks.json code:

```
            "command": "ng serve",
            "isBackground": true,
            "problemMatcher": [
                "$tsc-watch",
                {
                    "owner": "typescript",
                    "pattern": {
                        "regexp": "^([^\\s].*)\\((\\d+|\\,\\d+|\\d+,\\d+,\\d+,\\d+)\\):\\s+(error|warning|info)\\s+(TS\\d+)\\s*:\\s*(.*)$",
                        "file": 1,
                        "location": 2,
                        "severity": 3,
                        "code": 4,
                        "message": 5
                    },
                    "background": {
                        "activeOnStart": true,
                        "beginsPattern": "^\\s*\\d{1,2}:\\d{1,2}:\\d{1,2}(?: AM| PM)? - File change detected\\. Starting incremental compilation\\.\\.\\.",
                        "endsPattern": "^webpack: Compiled successfully."
                    }
                }
            ]
```

I haven't used that in a month or so because I'm on a new project. I tried to do the same thing as before, but it doesn't work.

My new code is:
```
        {
            "label": "ModelAgentFrontEnd",
            "identifier": "ModelAgentFrontEnd",
            "type": "shell",
            "command": "ng serve",
            "options": {
                "cwd": "../ModelAgentFrontEnd"
            },
            "presentation": {
                "reveal": "always",
                "panel": "new"
            },
            "isBackground": true,
            "problemMatcher": [
                "$tsc-watch",
                {
                    "owner": "typescript",
                    "pattern": {
                        "regexp": "^([^\\s].*)\\((\\d+|\\,\\d+|\\d+,\\d+,\\d+,\\d+)\\):\\s+(error|warning|info)\\s+(TS\\d+)\\s*:\\s*(.*)$",
                        "file": 1,
                        "location": 2,
                        "severity": 3,
                        "code": 4,
                        "message": 5
                    },
                    "background": {
                        "activeOnStart": true,
                        "beginsPattern": "^Your global Angular CLI version.*",
                        "endsPattern": "^webpack: Compiled successfully.*"
                    }
                }
            ]
        }
```

Any ideas as to why this doesn't work anymore? Did this stop working for anyone else?

**Comment by dbaeumer** [comment]
Actually no. How does the output of the task look like?

**Comment by BertoMejia** [comment]
It is a typical ng serve output. I didn't get fancy with regex at all, to make sure it works. The output is below:

```
Your global Angular CLI version (1.6.5) is greater than your local
version (1.6.0). The local Angular CLI version is used.

To disable this warning use "ng set --global warnings.versionMismatch=false".

@angular/compiler-cli@5.1.0 requires typescript@'>=2.4.2 <2.5.0' but 2.6.2 was found instead.
Using this version can result in undefined behaviour and difficult to debug problems.

Please run the following command to install a compatible version of TypeScript.

    npm install typescript@'>=2.4.2 <2.5.0'

To disable this warning run "ng set warnings.typescriptMismatch=false".

** NG Live Development Server is listening on localhost:4200, open your browser on http://localhost:4200/ **
Date: 2018-06-14T18:31:45.847Z
Hash: 1b793a4a1ae8dc08feec
Time: 11918ms
chunk {inline} inline.bundle.js (inline) 5.79 kB [entry] [rendered]
chunk {main} main.bundle.js (main) 657 kB [initial] [rendered]
chunk {polyfills} polyfills.bundle.js (polyfills) 607 kB [initial] [rendered]
chunk {styles} styles.bundle.js (styles) 363 kB [initial] [rendered]
chunk {vendor} vendor.bundle.js (vendor) 11.7 MB [initial] [rendered]

webpack: Compiled successfully.
```

**Comment by dbaeumer** [comment]
For this output `activeOnStart` needs to be false this the begin pattern is present even on the first run.

**Comment by kilianc** [comment]
I have the same problem with a Node.js app running inside a container locally. I have a bash script setting up the container and I can't attach the debugger using `preLaunchTask`. 

I am using:

```
,
      "problemMatcher": {
        "pattern": [{
          "regexp": "!!!DO NOT MATCH ME!!!",
          "file": 1,
          "location": 2,
          "message": 3
        }],
        "background": {
          "activeOnStart": true,
          "beginsPattern": "^.+Debugger listening.+",
          "endsPattern": "-----"
        }
      }
```

And my output is

```
[nodemon] 1.18.1
[nodemon] to restart at any time, enter `rs`
[nodemon] watching: *.*
[nodemon] starting `node --expose-gc --inspect=0.0.0.0 server.js || exit 1`
Debugger listening on ws://0.0.0.0:9229/ad551785-a8ff-47af-8e98-3e5890889205
For help, see: https://nodejs.org/en/docs/inspector
{
  "level": "info",
  "message": "server up @ http://localhost:5000"
}
```

When I change `beginsPattern` to something that doesn't match anything I get the Dialog with "Debug Anyway" when I do match something it just hangs.

If I run the task and then attach manually it works.

**Comment by dbaeumer** [comment]
@kilianc your ends pattern needs to match as well. This tells the debugger that the task actually has finished.

**Comment by kilianc** [comment]
@dbaeumer in my case is a server running, so end of the process is the end of the task. I don't log anything on exit but I only `^+C`.

I am confused on how the end of the task is going to affect the debugger attaching itself. Does the debugger wait for the task to be complete?

**Comment by kilianc** [comment]
@dbaeumer !!!!

```
"beginsPattern": "^Debugger listening.+$",
"endsPattern": "^For help.+$"
```

IT WORKS!

**Comment by synthkaf** [comment]
@dbaeumer seems like this hasn't moved in a while.

Is this feature still desired (I see it in the backlog) or should it be ditched entirely?

If this is something we'd like to have, is there a way we could break down the necessary steps to get it done?

**Comment by dbaeumer** [comment]
@synthkaf this is still on the list. That is why it is `On Deck`. The reason why it didn't get more tracking so far is that there is a reasonable workaround (a stupid problem pattern). 

But we just discussed this last week to decide what it means to make the pattern optional. Internally not a big issue but it has impact on the API as well.

**Comment by zgramana** [comment]
I'm able to use the workaround @skolmer outlines, but with a slight modification. You do not need to provide a pattern if you set `base`, as the following does:
```json
      {
        "label": "Start for Debugger",
        "type": "npm",
        "script": "start",
        "problemMatcher": [
          {
            "base": "$tsc-watch", // <-- extends the TypeScript watcher
            "background": {
              "activeOnStart": true,
              "beginsPattern": "Angular Live Development Server is listening",
              "endsPattern": "｢wdm｣: Compiled successfully"
            }
          }
        ],
        "isBackground": true,
        "group": "build"
      }
```
You still have to manually stop `ng serve` after disconnecting the debugger unless you add a `postDebugTask` to run a shell command to stop `ng serve`.

**Comment by clshortfuse** [comment]
This just broke for me. I haven't changed anything, but now the task is never signaled as complete:

launch.json
```
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Chrome Docs Debug",
      "type": "chrome",
      "request": "launch",
      "preLaunchTask": "npm: start:docs",
      "url": "http://localhost:8080/",
      "webRoot": "${workspaceFolder}/test",
      "sourceMapPathOverrides": {
        "webpack:///./*":"${workspaceRoot}/*",
      }
  },
  ]
}
```

tasks.json
```
{
  "version": "2.0.0",
  "tasks": [
    {
      "type": "npm",
      "script": "start:docs",
      "isBackground": true,
      "problemMatcher": [{
        "owner": "webpack",
        "source": "webpack",
        "fileLocation": "absolute",
        "severity": "error",
        "pattern": [{
          "regexp": "^ERROR in (.*)$",
        }, {
          "regexp": "^.*$"
        }, {
          "regexp": "^([^:]*): ([^:]*): ([^\\(]*) \\((\\d+):(\\d+)\\)$",
          "file": 2,
          "message": 3,
          "line": 4,
          "column": 5
        }],
        "background": {
          "activeOnStart": true,
          "beginsPattern": "^.*: Hash: *$",
          "endsPattern": "^.*(Compiled successfully)|(Failed to compile).*$"
        }
      }]
    }
  ]
}
```

Chrome Debugger doesn't trigger anymore because it thinks the task isn't complete. When I kill the task (with the trash icon), then I get the usual popup to continue debugging.

**Comment by clshortfuse** [comment]
Just realized, it's quite possibly from https://github.com/Microsoft/vscode/issues/65472

**Comment by tuath** [comment]
with watchers everywhere, i think a proper solution would be appropriate...

**Comment by JustinGrote** [comment]
Status on this? I want to set up a launch task to start azurite first and then start the azure functions host before launching the powershell debug, and I can't because of this issue if I make the functions host task dependson azurite.

I'm going to look into the biu program to "consolidate" it to one task maybe as a workaround.

**Comment by nevir** [comment]
It looks like the workarounds expressed here no longer work (seeing the same behavior as https://github.com/microsoft/vscode/issues/6209#issuecomment-449052178)

**Comment by dbaeumer** [comment]
@alexr00 can you please have a look. If you need explanation please step by.

**Comment by alexr00** [comment]
@dbaeumer I think I understand the issue. I'll check whether the problem matcher work around still works.

**Comment by simonhowey** [comment]
Running into this, the answer from Mike Wyatt worked for me:
https://stackoverflow.com/questions/44242048/how-to-make-vscode-not-wait-for-finishing-a-prelaunchtask

**Comment by alexr00** [comment]
I tested with this as a prelaunch task:
```
		{
			"type": "npm",
			"script": "watch",
			"problemMatcher": "$tsc-watch",
			"isBackground": true,
			"presentation": {
				"reveal": "never"
			},
			"group": {
				"kind": "build",
				"isDefault": true
			}
		}
```
The work around to use a watching problem matcher is still good.

**Comment by sinapis** [comment]
I have the same problem, trying to run nodemon as a prelaunch task. Workaround for now is to use the answer from this SO: https://stackoverflow.com/questions/44242048/how-to-make-vscode-not-wait-for-finishing-a-prelaunchtask

However, I just rely on the debugger attachment timeout to be long enough. Is there anyway to signal debugger to wait until nodemon actually starts node (there are build and lint tasks execution before nodemon, so timeout sometimes is not enough)?

**Comment by mmarczell-graphisoft** [comment]
Today all of the previously good workarounds have stopped working for me. No matter what I do I cannot get the debugging to start after a watching prelaunchTask.

**Comment by eathtespagheti** [comment]
Came here just to remind that none of the workarounds works as today

**Comment by DimaLapshyn** [comment]
> Came here just to remind that none of the workarounds works as today

@eathtespagheti Actually I've managed to set up a chain of two dependent background tasks with VS Code 1.64. The trick is to carefully write the `endsPattern` for the first task in the chain so it indeed matches the line indicating that the background service has started

**Comment by eathtespagheti** [comment]
> > Came here just to remind that none of the workarounds works as today
> 
> @eathtespagheti Actually I've managed to set up a chain of two dependent background tasks with VS Code 1.64. The trick is to carefully write the `endsPattern` for the first task in the chain so it indeed matches the line indicating that the background service has started

Can you post an example?

**Comment by DimaLapshyn** [comment]
> Can you post an example?

@eathtespagheti Sure:

### tasks.json

```
{
	"version": "2.0.0",
	"tasks": [
		{
			"type": "npm",
			"script": "testbed",
			"label": "npm: testbed",
			"detail": "node client-test-bed/serve.js",
			"isBackground": true,
			"dependsOn": [
				"npm: watch"
			],
			"problemMatcher": {
				"pattern": [
					{
						"regexp": ".",
						"file": 1,
						"location": 2,
						"message": 3
					}
				],
				"background": {
					"activeOnStart": true,
					"beginsPattern": ".",
					"endsPattern": "."
				}
			}
		},
		{
			"type": "npm",
			"script": "build",
			"group": "build",
			"problemMatcher": [],
			"label": "npm: build",
			"detail": "rollup -c"
		},
		{
			"type": "npm",
			"script": "watch",
			"group": "build",
			"problemMatcher": {
				"pattern": [
					{
						"regexp": ".",
						"file": 1,
						"location": 2,
						"message": 3
					}
				],
				"background": {
					"activeOnStart": true,
					"beginsPattern": "^\\[nodemon\\]\\s(starting|restarting)",
					"endsPattern": "^\\[nodemon\\]\\sclean\\sexit\\s\\-\\swaiting\\sfor\\schanges"
				}
			},
			"isBackground": true,
			"label": "npm: watch",
			"detail": "npm-watch"
		}
	]
}
```

### launch.json

```
{
  "configurations": [
    {
      "name": "Launch Test Bed in Chrome",
      "request": "launch",
      "type": "pwa-chrome",
      "url": "http://localhost:3000",
      "pathMapping": {
        "/": "${workspaceFolder}/client-test-bed",
        "/scripts": "${workspaceFolder}/dist/esm"
      },
      "preLaunchTask": "npm: testbed"
    }]
}
```

**Comment by Trass3r** [comment]
> Today all of the previously good workarounds have stopped working for me. No matter what I do I cannot get the debugging to start after a watching prelaunchTask.

If nothing else works you can use just `"isBackground": true` without a problem matcher which will prompt to debug anyway after a timeout. Or use a shell task with some magic like I explained in #151101.

**Comment by jknightdoeswork** [comment]
These problem matchers do not work for me. I'm trying to have a preLaunchTask start up a docker-compose. I prefer this to using the built in docker compose task because using a task produces the logs and all the output in the terminal window and the docker stuff can be shutdown by hitting CTRL+C in that terminal window. It's better. BUT, I cant use it this way because I cannot get the problem matcher to work. The problem matchers are insufficiently documented. The documentation doesnt even use the term 'regular expression'. It is unclear if there any constraints on the 'endsPattern'. It is unclear if there is any connection to the 'pattern.regexp', or 'pattern.file' etc.

It seems like a very simple feature for VSCode and something it should have. I'm dissapointed that it is so low priority.

My use case is this:

- docker-compose up debug-server
- attach remote debugger in vscode to the debug server

This seems like a very standard workflow.

All the related github issues around this topic show a consistent pattern of confused and dissapointed users. Saying things like "I got it to work" then "It doesnt work anymore".

Here's my attempt to write a 'endsPattern'. Honestly it's gross.

```		{
			"label": "Docker Compose Up Debug",
			"type": "shell",
			"isBackground": true,
			"command": "./scripts/dlv_start.sh",
			"problemMatcher": {
				"owner": "custom",
				"pattern": {
					"regexp": "__________________"
				},
				"background": {
					"activeOnStart": true,
					"beginsPattern": ".*",
					"endsPattern": ".*API server listening.*"
				}
			}
		},
```
